environment.yml

```diff
@@ -34,4 +34,6 @@ dependencies:
   - jasper!=1.900.31  # pinned NCL dependency
   - r-base
   - libunwind  # Needed for Python3.7+
+  - r-devtools
```

Comment:
R requirements should be added to esmvaltool/install/R/r_requirements.txt instead.

Reply:
I know, but I had to add them through conda-forge: lintr and devtools do not install if added through R install.packages ....

Reply:
By the way, I had to include the library "r-devtools" only in order to be able to use install_github() from R and install by hand the climdex.pcic.ncdf library.
We still have issue https://github.com/ESMValGroup/ESMValTool/issues/270#issuecomment-488992360 open .....

Reply:
Yes, that's another point we should work on. I noticed the [climdex.pcic](https://cran.r-project.org/web/packages/climdex.pcic/index.html) is actually available on CRAN. Would it also be possible to use just that? It's a bit sad that it is so difficult to get a package on CRAN. I'm wondering why they made two separate packages out of it, i.e. climdex.pcic and climdex.pcic.ncdf, but only managed to get one published.

Reply:
> Would it also be possible to use just that? 
I gave a look: not very practical in our case. The interface is very different and we would basically have to rewrite all of climdex.pcic.ncdf from scratch .... we are speaking about a lot of code to be added.
Actually these guys seem to be pretty committed to maintaining their tool (and they have for really many years now), so the github version is not going to disappear anytime soon. It seems as robust as accessing CRAN and I still think that one could make an exception in this case.

Reply:
> Why not? lintr installs just fine on CircleCI.

I tried again with a fresh conda environment. After installing environment.yml (without the two r-* packages) and trying to install lintr through r-requirements.txt, 
installation fails because installation of openssl fails. Here is an excerpt of the output:
```
installing to /home/jost/miniconda3/envs/esmvaltool8/lib/R/library/openssl/libs
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
Error: package or namespace load failed for ‘openssl’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/home/jost/miniconda3/envs/esmvaltool8/lib/R/library/openssl/libs/openssl.so':
  /home/jost/miniconda3/envs/esmvaltool8/lib/R/library/openssl/libs/openssl.so: undefined symbol: OpenSSL_add_all_digests
Error: loading failed
Execution halted
ERROR: loading failed
```

Reply:
It looks like it's looking for a function `OpenSSL_add_all_digests`, which is part of the `libcrypto.so` library provided by the openssl package, at least for version 1.0. In openssl version 1.1 it seems to have disappeared.

Strangely enough I also have a conda environment with openssl 1.1, in which lintr installed and works just fine. However, that is based on the conda-forge channel (without label/cf201901).

Reply:
The climdex.pcic package is what really calculates the indices. However it turns out to be very tedious and hard to use, when in most practical cases you just want all the indices calculated together for the dataset. To my understanding the climdex.pcic.ncdf started as a wrapper script to do this (also much more efficiently) and was made by an IT person who left for greener pastures quite a few years ago. Originally that wrapper would be a script that people would just have locally and circulate. (My boss tends to send people who want to know how to calculate the indices an old power point where this usage is demonstrated and it tends to lead to some confusion with recepients...) Anyways, that's way the two packages are treated so differently. I also did ask them about putting it on CRAN, however, I only got replys on how to upload the package for my own personal usage followed by silence.
All that said though, we REALLY do not want to reimplement the code to run without the climdex.pcic.ncdf package. This would require a massive amount of work, and most likely result in slower and possibly faulty results...

Reply:
I agree totally. These indices are tricky because behind their apparent simplicity in the definition, there are important technicalities to take into account. For comparing them it is really important that some standards are followed and these libraries really provide that .... We really should stick to using the climdex.pcic.ncdf library.